.azure-pipelines/client.yml

```diff
@@ -60,10 +57,66 @@ jobs:
       condition: and(succeededOrFailed(), ne(variables['Build.Reason'], 'PullRequest'))
       displayName: 'Component Detection'
 
-- job: 'Test'
+    - task: Maven@3
+      displayName: 'Generate project site'
+      inputs:
+        mavenPomFile: $(pomFile)
+        options: '--batch-mode'
+        mavenOptions: '-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn'
+        javaHomeOption: 'JDKVersion'
+        jdkVersionOption: '1.11'
+        jdkArchitectureOption: 'x64'
+        publishJUnitResults: false
+        goals: 'site'
 
-  dependsOn:
-  - 'Build'
+    - task: Maven@3
+      displayName: 'Install packages to local cache and generate SpotBugs report'
+      inputs:
+        mavenPomFile: $(pomFile)
+        options: '--batch-mode -Pspotbugs-reporting -DskipTests -Dgpg.skip'
+        mavenOptions: '-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn'
+        javaHomeOption: 'JDKVersion'
+        jdkVersionOption: '1.11'
+        jdkArchitectureOption: 'x64'
+        publishJUnitResults: false
+        goals: 'install'
+
+    - task: Maven@3
```

Comment:
@JonathanGiles @bsiegel is there any particular reason we split this up into multiple Maven tasks calls? Why not just run all the goals in the same call?